streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsMetadataState.java

```diff
@@ -296,9 +316,10 @@ private boolean hasPartitionsForAnyTopics(final List<String> topicNames, final S
     }
 
     private Set<String> getStoresOnHost(final Map<String, List<String>> storeToSourceTopics,
-        final Set<TopicPartition> sourceTopicPartitions, final String topologyName) {
+                                        final Set<TopicPartition> sourceTopicPartitions,
+                                        final String topologyName) {
         final InternalTopologyBuilder builder = topologyMetadata.lookupBuilderForNamedTopology(topologyName);
-        final Set<String> sourceTopicNames = builder.sourceTopicNames();
+        final Collection<String> sourceTopicNames = builder.fullSourceTopicNames();
```

Comment:
Just to clarify this is the actual fix right?

Reply:
One of them, yes